apps/nextjs/src/lib/provider/use-active-email-address.ts

```diff
@@ -0,0 +1,13 @@
+import { useParams } from "next/navigation";
+
+import { useEmailAddressById } from "@skylar/client-db";
+
+export const useActiveEmailAddress = () => {
+  const { providerIndex } = useParams();
+  let emailIdNumber =
+    parseInt(typeof providerIndex === "string" ? providerIndex : "1", 10) ?? 1;
+  if (emailIdNumber < 1) {
+    emailIdNumber = 1;
+  }
+  return useEmailAddressById(emailIdNumber);
```

Comment:
Throw here if the route does not exist?! the function should maybe not return undefined?! - because in every component its used you have extra logic if(!activeEmail)